lib/Factory/PostFactory.php

```diff
@@ -98,14 +98,17 @@ protected function get_post_class(WP_Post $post) : string {
 
 	protected function is_image(WP_Post $post) {
 		$src   = wp_get_attachment_url( $post->ID );
-		$check = wp_check_filetype( PathHelper::basename( $src ) );
+		$mimes = get_allowed_mime_types();
```

Comment:
I wonder whether we should use `wp_get_mime_types()` instead of `get_allowed_mime_types()`. I recently found a performance bottleneck with this when a lot images were processed with Timmy (https://github.com/mindkomm/timmy/commit/9812b9860bdd6f1500ad86685fc02b6c2794fefb).

The `get_allowed_mime_types()` includes a check with `user_can()`. But is this necessary for checking the file extension? I think we don’t need to check whether the current user is allowed to upload SVG images, that should have happened long before `is_image()` is called.

We already add the SVG mime type separately on the next line, so it actually doesn’t matter whether SVGs are allowed or not for the check whether an image is an SVG.

Also, quite a few plugins that manage the user rights to allow uploading SVG images hook into the `mime_types` filter, which is used by `wp_check_filetype()`. So maybe it’s worth to cache the result `wp_check_filetype()` so that we can make sure there’s no performance impact.

We could do this for all calls to `wp_check_filetype()` or `wp_check_filetype_and_ext()` in Timber, which would be a good separate issue.

Reply:
Awesome! didn't know about `wp_get_mime_types()` — that's def. the needed function (and why I didn't want to filter the `get_allowed_mime_types` which would interfere w the default security setting).

Agree w making the `wp_check_filetype()` caching a separate issue. For now this will allow 2.x to catch up w `master` (and a fresh reminder on why feature freeze should encompass even "innocent" ones like this)